week1/README.md

```diff
@@ -0,0 +1,220 @@
+
+Learn more about `JSX` here:
+
+- [Introducing JSX](https://reactjs.org/docs/introducing-jsx.html)
+
+## 3. Props
+
+### Unidrectional dataflow
+
+As we've learned in previous sections, in React data flows from component to component. However, this goes in an organized, tree-like manner. Take a look at the following diagram:
+
+![Component Hierarchy](./../assets/componenthierarchy.png)
+
+This is how we should think about creating our components: just like with the DOM tree we want to write components in a component tree. Here's how it could look in code:
+
+```js
+function App() {
+  return (
+    <Contacts>
+      // The Contacts component has 3 children: AddContact, ContactList and
+      SearchBar
+      <AddContact>
+        // This component has 1 child: AddContactForm
+        <AddContactForm />
+      </AddContact>
+      <ContactList>
+        // This component has 1 child: ContactCard
+        <ContactCard />
+      </ContactList>
+      // This component has no children, so is therefore self-closing
+      <SearchBar />
+    </Contacts>
+  );
+}
+```
+
+As you've learned previously, we use `props` to pass our `state` to our different components. But this doesn't just happen randomly; we do in from top to bottom, or in other words in a `unidirectional flow`. What does this mean? It means that `props` can only be passed **down** the component tree, but never up.
+
+In our example, it means that our `props` can only move from `<Contact>` > `<AddContact>`
+
+- [Unidirectional Data Flow in React](https://flaviocopes.com/react-unidirectional-data-flow/)
+
+## 4. Deploy a static site
+
+### What is a static site
+
+A static website is a collection of HTML, CSS and client-side JavaScript files. That's it. It is usually only a frontend, hosted on a web server that only serves the pages the client asks for (for example, `/`, `/about` or `/privacy`).
+
+As you've learned in the Node.js module, we can use server-side frameworks like `Express` or just native Node.js modules (like `http`) in order to serve our static files to a client.
+
+But what if we don't want to setup a whole web server to do this?
+
+### Static site generators
+
+We call these `static site generators`. They take a lot of the basic web server setup out of our hands. All we have to do is upload our frontend (in other words, our static website) to that particular service's web server and that's it. The generator creates a URL for us that we can use to access our website.
+
+Examples of static site generators are the following:
+
+- [Netlify](https://www.netlify.com/)
+- [Now](https://zeit.co/)
+- [Jekyll](https://jekyllrb.com/)
+
+To learn more about static site generators, study the following:
+
+- [The Reign of Static Site Generators](https://www.netlify.com/blog/2018/07/12/the-reign-of-static-site-generators/)
+- [Why use a Static Site Generator](https://www.youtube.com/watch?v=mDKQGANttyE)
+
```

Comment:
iirc, we discussed we should tell them about deployment tools now/netlify in first week so that they can deploy and submit homework as well